rust-doc-style-guide.md

```diff
@@ -0,0 +1,320 @@
+<!-- omit in toc -->
+# Rust Documentation Style Guide
```

Comment:
Do you think we can rename it to "Rust Doc Style Guide?"

It is much shorter and still intuitive enough.

Reply:
Sounds good to me, I dont really mind. 

Also, we could probably include the style guide from Rust book somewhere? https://github.com/rust-lang/book/blob/main/style-guide.md

Reply:
I think using the full word should always be preferable in documentation, personally.

Reply:
Let's use the full name, so there is no room for confusion. Change is now reverted!

Reply:
The reason is the general trend on the internet to shorten anything possible to make it "scannable." These days we have increasingly more abbreviations and abbreviations made of other abbreviations, such as AIoT.

But I agree that using full words is better, so let's stick with "documentation."

Reply:
@SergioGasquez 

> Also, we could probably include the style guide from Rust book somewhere? https://github.com/rust-lang/book/blob/main/style-guide.md

The style guide from Rust book is already included in the appendix under the name [book-ja](https://github.com/rust-lang-ja/book-ja/blob/master-ja/style-guide.md).

Please replace that link with the one you found.

Reply:
I've updated the links from book-ja (Japanese version of the book) to the English one and removed the one I included.